src/metabase/lib/schema.cljc

```diff
@@ -67,8 +82,9 @@
   [:fn
    {:error/message "Valid references for a single query stage"
     :error/fn      (fn [{:keys [value]} _]
-                     (ref-error-for-stage value))}
-   (complement ref-error-for-stage)])
+                     (when (map? value)
+                       (ref-error-for-stage value)))}
+   (every-pred map? (complement ref-error-for-stage))])
```

Comment:
Apparently when creating error messages with `me/humanize` Malli tries all of the schemas in an `:and`, even if the first one fails, so this is here to make sure we're handling values that aren't even maps at all when creating error messages.